agent/consul/rpc.go

```diff
@@ -257,6 +370,63 @@ func (s *Server) handleSnapshotConn(conn net.Conn) {
 	}()
 }
 
+func (s *Server) handleALPN_WANGossipPacketStream(conn net.Conn) error {
+	defer conn.Close()
+
+	transport := s.memberlistTransportWAN
+	for {
+		select {
+		case <-s.shutdownCh:
+			return nil
+		default:
+		}
+
+		// Note: if we need to change this format to have additional header
+		// information we can just negotiate a different ALPN protocol instead
+		// of needing any sort of version field here.
+		prefixLen, err := readUint32(conn, wanfed.GossipPacketMaxIdleTime)
```

Comment:
Oh one thing we should def do here for DOS prevention is enforce a sane max packet size.

If this len ends up being bigger than say 4MB (this is gRPC's default max RPC size that I proposed elsewhere adding to Consul RPC and plan to soon) we should just error the connection. I'm _pretty_ sure that's way bigger than Serf would ever need but we should probably check.

Even if we want to be super conservative and set it to 100MB or something I think a limit is better than no limit!

Reply:
4MB it is